.github/workflows/gsas2full.yml

```diff
@@ -1,16 +1,19 @@
-# Build the gsas2full self installer with Linux, MacOS-intel/ARM
-#
-#conda package on Mac (Intel) & Linux
+# Build the gsas2full self-installer for all supported platforms
 #
 name: build gsas2full
 
 #on: [push, pull_request, workflow_dispatch]   # why build on pull request?
```

Comment:
> Why build on pull request

Typically, you don't want to run the CI for every single commit that is pushed to the remote because there could be development commits that are part of unfinished features. Instead you reduce the scope to commits added to a specific branch where that branch is considered 'stable'
```yaml
on:
  push:
    branches:
      - 'main'
      - 'some-stable-branch'
```
But you want to know if new features break the CI before your merge them into the stable branch(es). So in addition to above you use the pull_request trigger.
```yaml
on:
  pull_request:
    types: [opened, reopened]
    branches:
      - 'main'
      - 'some-stable-branch'
```
This way the action is *also* run on the end of a branch that has a pull request into the stable branches.